tools/functional-tester/etcd-runner/command/global.go

```diff
@@ -61,16 +64,26 @@ func newClient(eps []string, timeout time.Duration) *clientv3.Client {
 	return c
 }
 
-func doRounds(rcs []roundClient, rounds int) {
-	var mu sync.Mutex
-	var wg sync.WaitGroup
+func doRounds(rcs []roundClient, rounds int, tokensPerRound int) {
+	var (
+		mu sync.Mutex
+		wg sync.WaitGroup
+	)
+
+	if rounds < 1 {
+		rounds = maxRounds
+	}
 
 	wg.Add(len(rcs))
 	finished := make(chan struct{})
+	limiter := rate.NewLimiter(rate.Limit(reqRate), reqRate)
 	for i := range rcs {
 		go func(rc *roundClient) {
 			defer wg.Done()
 			for rc.progress < rounds {
+				if err := limiter.WaitN(context.Background(), tokensPerRound); err != nil {
```

Comment:
I think this should be `tokensPerRound/len(rcs)` since there can be several roundClients working on a round